kinesis-produce-consume/src/main/java/demo/stream/OrdersSource.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package demo.stream;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.stereotype.Component;
+
+
+/**
+ * 
+ * @author Peter Oates
+ *
+ */
+@Component
+public class OrdersSource {
+
+	private final Log logger = LogFactory.getLog(getClass());
+	
+	private OrderProcessor orderOut;
+
+	@Autowired
+	public OrdersSource(OrderProcessor orderOut) {
+		this.orderOut = orderOut;
+	}
+
+	public void sendOrder(Event event) {
+
+		orderOut.ordersOut().send(MessageBuilder.withPayload(event).setHeader("contentType", "application/json")
+				.setHeader("aws_partitionKey", "100").build());
```

Comment:
I think we really should show the gain of the Binder:
1. I somehow believe that we don't need `contentType` header because you already have `ordersOut.content-type: application/json` binder property configured. Or is there some problem with that?
2. We have to move this `aws_partitionKey` to the `partitionKeyExpression` configuration property:
```
spring:
  cloud:
    stream:
      bindings:
        ordersOut: 
          destination: test_stream
          content-type: application/json
          producer:
            partitionKeyExpression: "1"
```
And finally this `send()` becomes very simple:
```
orderOut.ordersOut().send(new GenericMessage<>(event));
```

I'd even would go farther and really made this `OrdersSource` as a `@MessagingGateway` interface with the mapping to the `OrderProcessor.OUTPUT`:
```
@MessagingGateway
public interface OrdersSource {

    @Gateway(requestChannel = OrderProcessor.OUTPUT)
    void  sendOrder(Event event);
 
}
```

But I'm forced to agree that this would be an overhead for this sample as it is with JPA layer.